File: rxgooglemap/src/main/java/com/sdoward/rxgooglemap/InfoWindowCloseFunc.java
package com.sdoward.rxgooglemap;

import com.google.android.gms.maps.*;
import com.google.android.gms.maps.model.Marker;

import io.reactivex.Observable;
import io.reactivex.ObservableEmitter;
import io.reactivex.ObservableOnSubscribe;
import io.reactivex.functions.Function;

class InfoWindowCloseFunc implements Function<GoogleMap, Observable<Marker>> {

    @Override
    public Observable<Marker> apply(final GoogleMap googleMap) {
        return Observable.create(new ObservableOnSubscribe<Marker>() {
            @Override
            public void subscribe(final ObservableEmitter<Marker> emitter) {
                googleMap.setOnInfoWindowCloseListener(new GoogleMap.OnInfoWindowCloseListener() {
                    @Override
                    public void onInfoWindowClose(Marker marker) {
                        emitter.onNext(marker);
                    }
                });
            }
        });
    }
}

File: rxgooglemap/src/main/java/com/sdoward/rxgooglemap/InfoWindowCloseFunc.kt
package com.sdoward.rxgooglemap

import com.google.android.gms.maps.GoogleMap
import com.google.android.gms.maps.model.Marker
import io.reactivex.Observable
import io.reactivex.functions.Function

class InfoWindowCloseFunc : Function<GoogleMap, Observable<Marker>> {

    override fun apply(t: GoogleMap): Observable<Marker> {
        return Observable.create{ subscriber->
            t.setOnInfoWindowCloseListener {
                subscriber.onNext(it)
            }
        }
    }

}